nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024 Angsa Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__REMOVE_IN_COLLISION_GOALS_ACTION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__REMOVE_IN_COLLISION_GOALS_ACTION_HPP_
+
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "behaviortree_cpp/action_node.h"
+#include "nav2_msgs/srv/get_cost.hpp"
+
+namespace nav2_behavior_tree
+{
+
+class RemoveInCollisionGoals : public BT::ActionNodeBase
+{
+public:
+  typedef std::vector<geometry_msgs::msg::PoseStamped> Goals;
+
+  RemoveInCollisionGoals(
+    const std::string & xml_tag_name,
+    const BT::NodeConfiguration & conf);
+
+  /**
+   * @brief Function to read parameters and initialize class variables
+   */
+  void initialize();
+
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<Goals>("input_goals", "Original goals to remove from"),
+      BT::InputPort<std::string>("costmap",
+          "Which costmap to use for checking collision. Choices: local, global, both"),
```

Comment:
Is this alright or would you have it specified differently?

Reply:
Works for me, but maybe make it the actual `costmap_topic` to use instead - defaulted to global

Reply:
you mean the costmap service name right? e.g. `/global_costmap/get_cost_global_costmap`

Reply:
yeah!